### PR TITLE
Ftx usertrades streaming (With sub-account support)

### PR DESCRIPTION
**Pull request by imTigger**
Worked on top of PR #4075 by @makarid 
- Rebased on top of latest develop branch, and added sub-account support.
- Minor coding style fixes mentioned in original PR

Usage:
`ExchangeSpecification.setUserName` is used as sub-account, which is naturally null when not specified.

### CODE REVIEW

**Comment by makarid** [comment]
@imTigger thank you for this PR

File: xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingAdapters.java [hunk 1]
```diff
@@ -175,4 +180,22 @@ public static Ticker adaptTickerMessage(Instrument instrument, JsonNode jsonNode
                     .build())
         .collect(Collectors.toList());
   }
+
+  public static UserTrade adaptUserTrade(JsonNode jsonNode) {
+    return new UserTrade.Builder()
+            .currencyPair(new CurrencyPair(jsonNode.get("data").get("market").asText()))
+            .type(
+                    jsonNode.get("data").get("side").asText().equals("buy")
```
Comment (earce) [inline xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingAdapters.java:188]: string on left for comparison

Comment (imTigger) [reply xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingAdapters.java]: Yup, fixed

File: xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingService.java [hunk 2]
```diff
@@ -2,55 +2,147 @@
...
 
   public FtxStreamingService(String apiUrl) {
     super(apiUrl);
   }
 
+  public FtxStreamingService(String apiUrl, ExchangeSpecification exchangeSpecification) {
+    super(apiUrl);
+    this.exchangeSpecification = exchangeSpecification;
+  }
+
```
Comment (earce) [inline xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingService.java:38]: minor change but for consistency and to not have to store exchange spec in here, can you take a look at KrakenStreamingService and follow the pattern for creating the auth data, passing in a supplier?

Comment (imTigger) [reply xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingService.java:40]: 7de4dbb Just added a dto, used with supplier, avoid passing ExchangeSpecification into FtxStreamingService

**Pull request merged**